core/amber/src/main/python/pytexera/udf/udf_operator.py

```diff
@@ -80,6 +80,31 @@ def close(self) -> None:
         pass
 
 
+class UDFSourceOperator(SourceOperator):
+    def open(self) -> None:
+        """
+        Open a context of the operator. Usually can be used for loading/initiating some
+        resources, such as a file, a model, or an API client.
+        """
+        pass
+
+    @abstractmethod
+    def produce(self) -> Iterator[Optional[Union[TupleLike, TableLike]]]:
```

Comment:
Why is the function signature in the template
`def produce(self) -> Iterator[Union[TupleLike, TableLike, None]]` 
different than this one?

Why did you change from `Optional` to `None` in the template?

Reply:
The siguature should be `Iterator[Union[TupleLike, TableLike, None]]`, and it's been updated. They are logically equivalent, as `Union[X, None]` is equal to `Optional[X]`. see more [here](https://stackoverflow.com/questions/73893814/optional-typing-for-more-than-a-single-type)